### ISSUE TITLE
Lazy ExponentialMovingAverage for embedding matrix with sparse gradients

### ISSUE THREAD

**Issue by liuhu-bigeye** [tensorflow/tensorflow#29783]
Currently ExponentialMovingAverage is only designed for small Variables without sparse gradient(https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/python/training/moving_averages.py#L89). 

Directly calling "ema.apply(trainable_vars)" for embedding matrix is significantly slow. Would anyone provide a solution for sparse gradient in somewhat lazy manner?

Similar questions: https://github.com/pytorch/pytorch/issues/1285

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow-tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
